### ISSUE TITLE
Rerender introduces CONDA_BUILD_SYSROOT with OSX SDK on all systems

### ISSUE THREAD

**Issue by xhochy** [conda-forge/conda-forge.github.io#824]
See https://github.com/conda-forge/r-base-feedstock/pull/99/files#diff-69b02bc5032a05882001273f45aba3ab for a recent rerender. This makes it harder for me to locally build a package with the given pinnings as the OSX SDK is located somewhere else on my machine. This happened to work before in CI without this being set in the `.ci_support` directory.

What has changed? Which repo does contain this change? (I fail to find it in conda-smithy nor conda-forge-pinning).

**Comment by jakirkham** [comment]
Yeah the line comes from [conda-build itself]( https://github.com/conda/conda-build/blob/32a305208ffe53502e3a2bed8e83279da2cb5a5e/conda_build/variants.py#L36 ). So it is possible the change occurred there too. It would be worth checking with different versions of conda-build, and conda-smithy to see if there is a combination where it doesn't show up.

**Comment by jakirkham** [comment]
It's worth noting I'm not seeing this with other feedstocks that have been re-rendered recently. ( https://github.com/conda-forge/pip-feedstock/pull/31 ) So there may be something unique to that feedstock as well.

**Comment by pkgw** [comment]
I just ran into this issue in `poppler-feedstock` and it wasted me a *lot* of time trying to figure out why the Conda compilers all stopped working inside builds.

My workaround was just to create a symlink at the magic path of `/opt/MacOSX10.10.sdk` ... gross and not fit for production, but enough for local testing.

**Comment by isuruf** [comment]
PRs welcome to conda-smithy to remove `CONDA_BUILD_SYSROOT` from used variables.
